rls/src/main/java/io/grpc/rls/CachingRlsLbClient.java

```diff
@@ -218,13 +220,14 @@ public void onNext(io.grpc.lookup.v1.RouteLookupResponse value) {
               public void onError(Throwable t) {
                 logger.log(ChannelLogLevel.DEBUG, "Error looking up route:", t);
                 response.setException(t);
-                throttler.registerBackendResponse(false);
+
+                throttler.registerBackendResponse(getCause(t) instanceof ThrottledException);
```

Comment:
You can just call `true` here. If you look at comment on `AdaptiveThrottler.throttledStat` which this boolean increments:

> Counter for the total number of requests that were throttled by either the client (this class) or the backend in recent history. 

When we get an error here you are registering that the backend has "throttled" and you don't care if `AdaptiveThrottler` has thrown `ThrottledException`. It increments client side throttles when this class calls `shouldThrottle()`.

Reply:
Isn't there a chance that onError got called for other reasons than throttling that you would then be attributing to throttling?

Reply:
This will _never_ be `ThrottledException`. The server can't send that. A test with inprocess transport would demonstrate that.

The point of adaptive throttling is to fail requests early if they were going to fail anyway, so virtually any failure should probably be included. go/grpc-rls-lb-policy-design#heading=h.ngvrl0gh1gjs doesn't list out specific failures to consider failures; it only says "successful" and "total number of requests", and there was clearly a request here but it sounds strange to consider it successful.